### PR TITLE
Make WrappedFunction curry type info

### PR DESCRIPTION
**Pull request by inlined**
Sorry that the formatter made a _bunch_ of changes. The majority of the impactful changes here are that the wrapped function types keep the type info of the functions that they wrap. This also required us a fix to the `CloudFunction` type because it is now generic based on an EventType rather than event data so that we can keep type info for subtypes of CloudEvent.

I'd like to eventually look at refactoring this to have fewer places where we have `any`. I think we might want to have three pieces:

`function baseEvent<T>(): Omit<CloudEvent<any>, 'data'>` that returns what you have today. Then in each partial you do the combination of `baseEvent` with per-event type overrides (e.g. bucket for Storage) and the partial data. By putting the event construction lower in the stack we can make sure data is referentially correct (e.g. the StorageEvent.bucket matches the StorageEvent.data.bucket)

### CODE REVIEW

File: src/cloudevent/generate.ts [hunk 1]
```diff
@@ -1,46 +1,62 @@
...
+    ...generateMockCloudEventPartial(cloudFunction),
   };
 }
 
 /** @internal */
-function generateBaseCloudEvent<EventType>(cloudFunction: CloudFunction<EventType>): CloudEvent {
+function generateBaseCloudEvent<EventType extends CloudEvent<unknown>>(
+  cloudFunction: CloudFunction<EventType>
+): EventType {
+  // TODO: Consider refactoring so that we don't use this utility function. This
```
Comment (TheIronDev) [inline src/cloudevent/generate.ts:37]: 👍

File: src/cloudevent/partials/partials.ts [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
+import { alertsCrashlyticsOnNewNonfatalIssuePublished } from './alerts/crashlytics-on-new-nonfatal-issue-published';
+import { alertsCrashlyticsOnRegressionAlertPublished } from './alerts/crashlytics-on-regression-alert-published';
+import { alertsCrashlyticsOnStabilityDigestPublished } from './alerts/crashlytics-on-stability-digest-published';
+import { alertsCrashlyticsOnVelocityAlertPublished } from './alerts/crashlytics-on-velocity-alert-published';
+import { alertsAppDistributionOnNewTesterIosDevicePublished } from './alerts/app-distribution-on-new-tester-ios-device-published';
+import { alertsBillingOnPlanAutomatedUpdatePublished } from './alerts/billing-on-plan-automated-update-published';
+import { alertsBillingOnPlanUpdatePublished } from './alerts/billing-on-plan-update-published';
+import { eventarcOnCustomEventPublished } from './eventarc/eventarc-on-custom-event-published';
+import { pubsubOnMessagePublished } from './pubsub/pubsub-on-message-published';
+import { storageV1 } from './storage';
```
Comment (TheIronDev) [inline src/cloudevent/partials/partials.ts:14]: Shouldn't there be a new file here?

File: src/main.ts [hunk 3]
```diff
@@ -53,20 +45,20 @@ export {
...
-): WrappedScheduledFunction | WrappedFunction | WrappedV2Function {
-  if (isV2CloudFunction(cloudFunction)) {
-    return wrapV2<T>(cloudFunction as CloudFunctionV2<T>);
+  cloudFunction: CloudFunctionV1<T>
+): WrappedScheduledFunction | WrappedFunction<T>;
+export function wrap<T extends CloudEvent<unknown>>(
+  cloudFunction: CloudFunctionV2<T>
+): WrappedV2Function<T>;
+
+export function wrap<T, V extends CloudEvent<unknown>>(
```
Comment (TheIronDev) [inline src/main.ts:57]: I like this function overloading much better than what I did earlier 🐢

**Review by TheIronDev — APPROVED** [review]

File: src/cloudevent/partials/partials.ts (hunk 2 above)
Comment (TheIronDev) [reply src/cloudevent/partials/partials.ts:14]: src/cloudevent/partials/partials.ts:14:27 - error TS2307: Cannot find module './storage' or its corresponding type declarations.

14 import { storageV1 } from './storage';

Comment (inlined) [reply src/cloudevent/partials/partials.ts:14]: Indeed. Added missing file back.

**Pull request merged**